### DISCUSSION TITLE
Does the ConfigureAwait really matter?

### DISCUSSION THREAD

**Discussion by doggy8088** [testcontainers/testcontainers-dotnet#1000] [Q&A]
I see almost all the examples on TestContiners docs are append `.ConfigureAwait(false)` to all Tasks. For example:

```cs
await Image.InitializeAsync()
  .ConfigureAwait(false);

await _weatherForecastNetwork.CreateAsync()
  .ConfigureAwait(false);

await _msSqlContainer.StartAsync()
  .ConfigureAwait(false);

await _weatherForecastContainer.StartAsync()
  .ConfigureAwait(false);
```

Is that really matter to the performance?

**Comment by HofmeisterAn** [comment]
Well, it may look cumbersome, but we do not need to return to the calling thread and wait until it is ready. I am just used to set it everywhere (expect UI, etc.). [Here](https://www.youtube.com/watch?v=J0mcYVxJEl0) is a good talk that covers the topic as well.

**Reply by doggy8088** [reply]
I know what it is. I'm just thinking that might not a big deal. The threading issue seems a little bit overtuning.
